### PR TITLE
[ONNX] Suppport torch.dot and torch.nn.utils.spectral_norm

### LINKED ISSUE #27723: ONNX export failed on ATen operator mv
## 🐛 Bug
I run this code on [Colab](https://colab.research.google.com/drive/1zRUeHSid0PEbZULODFowxOqxc5R6dvja) 
Running torch.onnx.export gives this error.

> /content/DeOldify/deoldify/unet.py:56: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
>   if ssh != up_out.shape[-2:]:
> /usr/local/lib/python3.6/dist-packages/torch/onnx/utils.py:562: UserWarning: ONNX export failed on ATen operator mv because torch.onnx.symbolic_opset9.mv does not exist
>   .format(op_name, opset_version, op_name))
> 
> 
> KeyError                                  Traceback (most recent call last)
> <ipython-input-11-6588fcd71ffa> in <module>()
>       1 dummy_input = torch.randn(1, 3, 224, 224)
>       2 
> ----> 3 torch.onnx.export(learn.model.cpu(), dummy_input, "colorizer.onnx")
> 
> 7 frames
> /usr/local/lib/python3.6/dist-packages/torch/onnx/symbolic_registry.py in get_registered_op(opname, domain, version)
>      89         warnings.warn("ONNX export failed. The ONNX domain and/or version are None.")
>      90     global _registry
> ---> 91     return _registry[(domain, version)][opname]
> 
> KeyError: 'mv'

## To Reproduce

1.
Running this code

```
!git clone https://github.com/jantic/DeOldify.git DeOldify 

cd DeOldify

!pip install -r requirements.txt

import fastai
from deoldify.visualize import *

torch.backends.cudnn.benchmark = True
!mkdir 'models'
!wget https://www.dropbox.com/s/zkehq1uwahhbc2o/ColorizeArtistic_gen.pth?dl=0 -O ./models/ColorizeArtistic_gen.pth

from deoldify.generators import gen_inference_deep, gen_inference_wide

root_folder:Path=Path('./')
weights_name:str='ColorizeArtistic_gen'

learn = gen_inference_deep(root_folder=root_folder, weights_name=weights_name)

dummy_input = torch.randn(1, 3, 224, 224)

torch.onnx.export(learn.model.cpu(), dummy_input, "colorizer.onnx")

```

## Expected behavior

I expect the conversion to Onnx.

## Environment

 - PyTorch Version:  1.2.0
 - OS : Linux
 - How you installed PyTorch:pip
 - Python version:3.6

### PR DESCRIPTION
**Pull request by BowenBao**
Fixes #27723

### CODE REVIEW

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 1]
```diff
@@ -5707,6 +5707,27 @@ def forward(self, input, other):
         y = torch.randint(10, (5, ))
         self.run_test(MatmulModel(), (x, y))
 
+    @skipIfUnsupportedMinOpsetVersion(9)  # MatMul long inputs is added in ONNX opset 9.
+    def test_dot(self):
+        class MatmulModel(torch.nn.Module):
+            def forward(self, input, other):
+                return torch.dot(input, other)
```
Comment (vadimkantorov) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:5714]: are `torch.dot` and `torch.matmul` aliases?

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 2]
```diff
@@ -5707,6 +5707,27 @@ def forward(self, input, other):
...
+        y = torch.randn(5, requires_grad=True)
+        self.run_test(MatmulModel(), (x, y))
+
+        x = torch.randint(10, (5, ))
+        y = torch.randint(10, (5, ))
+        self.run_test(MatmulModel(), (x, y))
+
+    @disableScriptTest()  # SpectralNorm not TorchScript compatible.
+    def test_spectral_norm(self):
+        m = torch.nn.utils.spectral_norm(torch.nn.Linear(2, 4))
```
Comment (jiafatom) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:5726]: What is `torch.nn.utils.spectral_norm` look like in torchscript? Is it composed of several Aten operators and one of them is `dot`?

Comment (BowenBao) [reply test/onnx/test_pytorch_onnx_onnxruntime.py:5726]: src: https://github.com/pytorch/pytorch/blob/master/torch/nn/utils/spectral_norm.py SpectralNorm is not nn.Module, just a regular python class.

**Review by jiafatom — APPROVED** [review]
Thanks, LGTM!

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 3]
```diff
@@ -5707,6 +5707,27 @@ def forward(self, input, other):
         y = torch.randint(10, (5, ))
         self.run_test(MatmulModel(), (x, y))
 
+    @skipIfUnsupportedMinOpsetVersion(9)  # MatMul long inputs is added in ONNX opset 9.
+    def test_dot(self):
+        class MatmulModel(torch.nn.Module):
+            def forward(self, input, other):
+                return torch.dot(input, other)
+
+        x = torch.randn(5, requires_grad=True)
```
Comment (fatcat-z) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:5716]: Just curious, why do we need to set requires_grad=True?

Comment (BowenBao) [reply test/onnx/test_pytorch_onnx_onnxruntime.py:5716]: could be potential use case. I mimicked the other Matmul test case.

**Pull request merged**